### PR TITLE
Retry if DNS resolution fails

### PR DESCRIPTION
**Pull request by intelliot**
## High Level Overview of Change

Currently if DNS resolution fails, the app gives up. Instead we should retry.

This also adds a relevant timestamp to one of the logs.

### Context of Change

Example logs:
```
Seconds per ledger: 5
Connection timed out: ws://<ip>:51233. Terminating.
[mainnet] [ws://<ip>:51233] Connection closed. 1 connections remaining.
[mainnet] ERROR: Error: queryA ECONNREFUSED r.ripple.com (Failed to resolve r.ripple.com)
(node:23437) UnhandledPromiseRejectionWarning: undefined
(node:23437) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:23437) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Connection timed out: ws://<ip>:51233. Terminating.
[mainnet] [ws://<ip>:51233] Connection closed. 0 connections remaining.
[mainnet] ERROR: Error: queryA ECONNREFUSED r.ripple.com (Failed to resolve r.ripple.com)
(node:23437) UnhandledPromiseRejectionWarning: undefined
(node:23437) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
[WARN]  WebClient:0 http request failed getaddrinfo ENOTFOUND slack.com
[WARN]  WebClient:0 http request failed getaddrinfo ENOTFOUND slack.com
[WARN]  WebClient:0 http request failed getaddrinfo ENOTFOUND slack.com
Seconds per ledger: Infinity
[WARN]  WebClient:0 http request failed getaddrinfo ENOTFOUND slack.com
[WARN]  WebClient:0 http request failed getaddrinfo ENOTFOUND slack.com
Seconds per ledger: Infinity
[mainnet] loadUNL error: FetchError: request to https://vl.ripple.com/ failed, reason: getaddrinfo ENOTFOUND vl.ripple.com
    at ClientRequest.<anonymous> (/Users/user/ripple/validation-count-reporter/node_modules/node-fetch/lib/index.js:1455:11)
    at ClientRequest.emit (events.js:223:5)
    at ClientRequest.EventEmitter.emit (domain.js:475:20)
    at TLSSocket.socketErrorListener (_http_client.js:406:9)
    at TLSSocket.emit (events.js:223:5)
    at TLSSocket.EventEmitter.emit (domain.js:475:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:81:21) {
  message: 'request to https://vl.ripple.com/ failed, reason: getaddrinfo ENOTFOUND vl.ripple.com',
  type: 'system',
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND'
}
Seconds per ledger: Infinity
[WARN]  WebClient:0 http request failed getaddrinfo ENOTFOUND slack.com
Seconds per ledger: Infinity
Seconds per ledger: Infinity
[mainnet] loadUNL error: FetchError: request to https://vl.ripple.com/ failed, reason: getaddrinfo ENOTFOUND vl.ripple.com
    at ClientRequest.<anonymous> (/Users/user/ripple/validation-count-reporter/node_modules/node-fetch/lib/index.js:1455:11)
    at ClientRequest.emit (events.js:223:5)
    at ClientRequest.EventEmitter.emit (domain.js:475:20)
    at TLSSocket.socketErrorListener (_http_client.js:406:9)
    at TLSSocket.emit (events.js:223:5)
    at TLSSocket.EventEmitter.emit (domain.js:475:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:81:21) {
  message: 'request to https://vl.ripple.com/ failed, reason: getaddrinfo ENOTFOUND vl.ripple.com',
  type: 'system',
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND'
}
[WARN]  WebClient:0 http request failed getaddrinfo ENOTFOUND slack.com
Seconds per ledger: Infinity
Seconds per ledger: Infinity
Seconds per ledger: Infinity
[mainnet] loadUNL error: FetchError: request to https://vl.ripple.com/ failed, reason: getaddrinfo ENOTFOUND vl.ripple.com
    at ClientRequest.<anonymous> (/Users/user/ripple/validation-count-reporter/node_modules/node-fetch/lib/index.js:1455:11)
    at ClientRequest.emit (events.js:223:5)
    at ClientRequest.EventEmitter.emit (domain.js:475:20)
    at TLSSocket.socketErrorListener (_http_client.js:406:9)
    at TLSSocket.emit (events.js:223:5)
    at TLSSocket.EventEmitter.emit (domain.js:475:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:81:21) {
  message: 'request to https://vl.ripple.com/ failed, reason: getaddrinfo ENOTFOUND vl.ripple.com',
  type: 'system',
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND'
}
Seconds per ledger: Infinity
Seconds per ledger: Infinity
[WARN]  WebClient:0 http request failed getaddrinfo ENOTFOUND slack.com
```

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## Test Plan

Set the `SLACK_TOKEN` env var and then run the bot with `yarn start`

### CODE REVIEW

File: src/collector.ts [hunk 1]
```diff
@@ -199,7 +199,11 @@ export class Collector {
         const validatorList = JSON.parse(buffer.toString('ascii'))
         if (validatorList.sequence <= getLastValidatorListSequence()) {
           console.info(
-            `[${this.network}] Using validator list #${validatorList.sequence}.`,
+            `${new Date().toLocaleString('en-US', {
+              timeZone: 'America/Los_Angeles',
```
Comment (hbergren) [inline src/collector.ts]: Why `America/Los_Angeles` over `UTC`?

Is this logged to Slack, or just locally?

File: src/collector.ts [hunk 2]
```diff
@@ -365,10 +369,14 @@ export class Collector {
...
       dns.resolve4(hostname, (err, ips) => {
         if (err) {
+          // Example:
+          //   [mainnet] ERROR: Error: queryA ECONNREFUSED r.ripple.com (Failed to resolve r.ripple.com)
           console.error(
             `[${this.network}] ERROR: ${err} (Failed to resolve ${hostname})`,
           )
-          return reject()
+          // Important: Try again
+          setTimeout(() => this.connectToRippled(retryCount + 1), 1000)
```
Comment (hbergren) [inline src/collector.ts]: So this tries to reconnect once per second?

File: src/collector.ts (hunk 1 above)
Comment (intelliot) [reply src/collector.ts]: This uses console.info which only logs locally

Comment (intelliot) [reply src/collector.ts]: Timezone doesn’t matter but for convenience I used the timezone I’m in. Feel free to change it to UTC if you prefer.

File: src/collector.ts (hunk 2 above)
Comment (intelliot) [reply src/collector.ts]: Yes

**Pull request merged**